src/Helpers/IPFSClient.ts

```diff
@@ -0,0 +1,98 @@
+import * as FormData from 'form-data'
+import fetch from 'node-fetch'
+import * as str from 'string-to-stream'
+
+interface IPFSConfiguration {
+  readonly ipfsUrl: string
+}
+
+interface FetchConfiguration {
+  readonly timeout: number
+}
+
+enum Type {
+  File = 'File',
+  Directory = 'Directory',
+}
+
+interface IPFSObject {
+  readonly Name?: string
+  readonly Hash: string
+  readonly Size?: number
+  readonly Type?: Type
+  readonly Links?: ReadonlyArray<IPFSObject>
+}
+
+interface LSResult {
+  Arguments: {
+    [key: string]: string
+  }
+  Objects: {
+    [key: string]: IPFSObject
+  }
+}
+
+type addFileToDirectory = (directoryHash: string, fileName: string, filehash: string) => Promise<IPFSObject>
+
+type addText = (text: string, x?: FetchConfiguration) => Promise<IPFSObject>
+
+type cat = (s: string, x?: FetchConfiguration) => Promise<string>
+
+type createEmptyDirectory = () => Promise<IPFSObject>
+
+type getDirectoryFileHashes = (s: string) => Promise<ReadonlyArray<string>>
+
+type ls = (s: string) => Promise<LSResult>
+
+export class IPFS {
+  private readonly url: string
+
+  constructor(configuration: IPFSConfiguration) {
+    this.url = configuration.ipfsUrl
+  }
+
+  addFileToDirectory: addFileToDirectory = async (directoryHash, fileName, fileHash) => {
+    const response = await fetch(
+      `${this.url}/api/v0/object/patch/add-link?arg=${directoryHash}&arg=${fileName}&arg=${fileHash}`
+    )
+    return await response.json()
+  }
+
+  addText: addText = async (text: string, fetchConfiguration) => {
+    const formData = new FormData() // { maxDataSize: 20971520 }
```

Comment:
Does converting the text to form data belong in this method? or is it the callers responsibility?

Reply:
I think having a layer that abstracts this away makes sense. In my mind, the IPFSClient should hide away any details of _how_ to communicate with the API.